### PR TITLE
[ENH] allow passing of pretrained models to hugging face forecaster

### PR DESCRIPTION
**Pull request by Spinachboul**
#### Reference Issues/PRs
Fixes https://github.com/sktime/sktime/issues/6561

#### What does this implement/fix? Explain your changes.
This code adds the functionality of invoking the model object along with the model path giving the end users flexibility of customizing their models, perform HPO and fine tuning the parameters.

#### Does your contribution introduce a new dependency? If yes, which one?
No
#### What should a reviewer concentrate their feedback on?

The reviewer could give their feedback upon the code robustness and its feasibility, keeping in mind the code structure and dependencies.

#### Did you add any tests for the change?

No currently! I would add some tests upon further review of the maintainers.

#### Any other comments?
No

### CODE REVIEW

File: sktime/forecasting/hf_transformers_forecaster.py [hunk 1]
```diff
@@ -36,9 +31,9 @@ class HFTransformersForecaster(BaseForecaster):
 
     Parameters
     ----------
-    model_path : str
-        Path to the huggingface model to use for forecasting. Currently,
-        Informer, Autoformer, and TimeSeriesTransformer are supported.
```
Comment (fkiraly) [inline sktime/forecasting/hf_transformers_forecaster.py:41]: we should give a list of supported str or arcihtectures.

**Review by fkiraly — CHANGES_REQUESTED** [review]
Nice, thanks for the contribution! Much requested.

FYI @benHeid for review.

Major comment: we should ensure we test the new feature - we can use `get_test_params` which specify test parameter sets, or a separate `pytest` test.

I also left a minor comment related to the docstring, we should:

* specify the expected format of the path `str`
* specify which architectures are supported by the adapter - not all hf forecasters follow the former architecture

**Comment by fkiraly** [comment]
tests are failing code quality, see here for how to fix:
https://www.sktime.net/en/stable/developer_guide/coding_standards.html

File: sktime/forecasting/hf_transformers_forecaster.py (hunk 1 above)
Comment (Spinachboul) [reply sktime/forecasting/hf_transformers_forecaster.py:41]: Got it! I will mention the architectures currently supported by sktime

**Comment by Spinachboul** [comment]
@fkiraly & @benHeid 
I am planning to add these architectures as the supported ones for now
```{python}
SUPPORTED_ARCHITECTURES = [
        "Informer",
        "Autoformer",
        "TimeSeriesTransformer"
    ]
```

If any other ones are supported please do inform!!

**Comment by Spinachboul** [comment]
```{logs}
PS C:\Users\mridu\Desktop\Sktime\sktime\forecasting\personal> pytest
f"C:\Users\{sername}\AppData\Local\Programs\Python\Python311\Lib\site-packages\_pytest\config\__init__.py:328:" PluggyTeardownRaisedWarning: 
A plugin raised an exception during an old-style hookwrapper teardown.
Plugin: helpconfig, Hook: pytest_cmdline_parse
UsageError: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --timeout -n auto
  inifile: C:\Users\{username}\Desktop\Sktime\setup.cfg
  rootdir: C:\Users\{username}\Desktop\Sktime
For more information see https://pluggy.readthedocs.io/en/stable/api_reference.html#pluggy.PluggyTeardownRaisedWarning
  config = pluginmanager.hook.pytest_cmdline_parse(
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --timeout -n auto        
  inifile: C:\Users\{username}\Desktop\Sktime\setup.cfg
  rootdir: C:\Users\{username}\Desktop\Sktime
```
These are the error logs which I am getting when trying to test the functions using `pytest` package
But the issue might due to the presence of `setup.cfg` present in the root folder

@fkiraly abd @benHeid Any suggestions?

**Comment by fkiraly** [comment]
I think you haven´t installed the `dev` dependency set, which installs `pytest-timeout` required for this flag?

**Comment by Spinachboul** [comment]
> I think you haven´t installed the `dev` dependency set, which installs `pytest-timeout` required for this flag?

Yeah! ur right! Just did it now and it works. Thanks!

**Comment by Spinachboul** [comment]
@fkiraly  and @benHeid 
Previously I addded a textual HF model but after shifting to the a forecasting HF model, I am able to pass all the tests

Here is my console logs:
```
======================================================== test session starts ========================================================
platform win32 -- Python 3.11.9, pytest-7.3.1, pluggy-1.5.0
Matplotlib: 3.7.1
Freetype: 2.6.1
rootdir: C:\Users\mridu\Desktop\Sktime
configfile: setup.cfg
plugins: anyio-3.7.1, dash-2.14.2, hypothesis-6.103.1, cov-5.0.0, mpl-0.17.0, timeout-2.3.1, xdist-3.6.1
timeout: 600.0s
timeout method: thread
timeout func_only: False
4 workers [1 item]     
C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\coverage\inorout.py:519: CoverageWarning: Module sktime was previously imported, but not measured (module-not-measured)
  self.warn(msg, slug="module-not-measured")
C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\coverage\control.py:888: CoverageWarning: No data was collected. (no-data-collected)
  self._warn("No data was collected.", slug="no-data-collected")
C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\coverage\inorout.py:519: CoverageWarning: Module sktime was previously imported, but not measured (module-not-measured)
  self.warn(msg, slug="module-not-measured")
C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\coverage\inorout.py:519: CoverageWarning: Module sktime was previously imported, but not measured (module-not-measured)
  self.warn(msg, slug="module-not-measured")
C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\coverage\control.py:888: CoverageWarning: No data was collected. (no-data-collected)
  self._warn("No data was collected.", slug="no-data-collected")
C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\coverage\control.py:888: CoverageWarning: No data was collected. (no-data-collected)
  self._warn("No data was collected.", slug="no-data-collected")
.                                                                                                                                                                              [100%]C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\coverage\inorout.py:519: CoverageWarning: Module sktime was previously imported, but not measured (module-not-measured)
  self.warn(msg, slug="module-not-measured")

================================================================================= warnings summary ==================================================================================
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\tensorflow\python\debug\cli\debugger_cli_common.py:19
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\tensorflow\python\debug\cli\debugger_cli_common.py:19
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\tensorflow\python\debug\cli\debugger_cli_common.py:19
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\tensorflow\python\debug\cli\debugger_cli_common.py:19
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\tensorflow\python\debug\cli\debugger_cli_common.py:19: DeprecationWarning: module 'sre_constants' is deprecated
    import sre_constants

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\torch\utils\tensorboard\__init__.py:4
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\torch\utils\tensorboard\__init__.py:4
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\torch\utils\tensorboard\__init__.py:4
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\torch\utils\tensorboard\__init__.py:4
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\torch\utils\tensorboard\__init__.py:4: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    if not hasattr(tensorboard, "__version__") or LooseVersion(

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\torch\utils\tensorboard\__init__.py:6
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\torch\utils\tensorboard\__init__.py:6
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\torch\utils\tensorboard\__init__.py:6
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\torch\utils\tensorboard\__init__.py:6
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\torch\utils\tensorboard\__init__.py:6: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    ) < LooseVersion("1.15"):

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\utils\requirements_utils.py:20
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\utils\requirements_utils.py:20
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\utils\requirements_utils.py:20
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\utils\requirements_utils.py:20
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\utils\requirements_utils.py:20: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources  # noqa: TID251

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('mpl_toolkits')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('google')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('sphinxcontrib')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\pkg_resources\__init__.py:2825: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('zope')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
    declare_namespace(pkg)

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:63
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:63
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:63
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:63
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:63: PydanticDeprecatedSince20: Pydantic V1 style `@validator` validators are deprecated. You should migrate to Pydantic V2 style `@field_validator` validators, see the migration guide for more details. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.7/migration/
    @validator("cohere_api_key", pre=True)

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:72
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:72
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:72
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:72
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:72: PydanticDeprecatedSince20: Pydantic V1 style `@validator` validators are deprecated. You should migrate to Pydantic V2 style `@field_validator` validators, see the migration guide for more details. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.7/migration/
    @validator("ai21labs_api_key", pre=True)

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:82
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:82
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:82
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:82
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:82: PydanticDeprecatedSince20: Pydantic V1 style `@validator` validators are deprecated. You should migrate to Pydantic V2 style `@field_validator` validators, see the migration guide for more details. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.7/migration/
    @validator("mosaicml_api_key", pre=True)

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:113
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:113
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:113
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:113
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:113: PydanticDeprecatedSince20: Pydantic V1 style `@validator` validators are deprecated. You should migrate to Pydantic V2 style `@field_validator` validators, see the migration guide for more details. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.7/migration/
    @validator("openai_api_key", pre=True)

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:168
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:168
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:168
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:168
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:168: PydanticDeprecatedSince20: Pydantic V1 style `@validator` validators are deprecated. You should migrate to Pydantic V2 style `@field_validator` validators, see the migration guide for more details. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.7/migration/
    @validator("anthropic_api_key", pre=True)

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:177
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:177
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:177
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:177
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:177: PydanticDeprecatedSince20: Pydantic V1 style `@validator` validators are deprecated. You should migrate to Pydantic V2 style `@field_validator` validators, see the migration guide for more details. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.7/migration/
    @validator("palm_api_key", pre=True)

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:282
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:282
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:282
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:282
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:282: PydanticDeprecatedSince20: Pydantic V1 style `@validator` validators are deprecated. You should migrate to Pydantic V2 style `@field_validator` validators, see the migration guide for more details. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.7/migration/
    @validator("provider", pre=True)

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:303
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:303
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:303
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:303
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:303: PydanticDeprecatedSince20: Pydantic V1 style `@validator` validators are deprecated. You should migrate to Pydantic V2 style `@field_validator` validators, see the migration guide for more details. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.7/migration/
    @validator("config", pre=True)

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:320
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:320
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:320
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:320
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:320: PydanticDeprecatedSince20: Pydantic V1 style `@validator` validators are deprecated. You should migrate to Pydantic V2 style `@field_validator` validators, see the migration guide for more details. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.7/migration/
    @validator("name")

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:330
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:330
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:330
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:330
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:330: PydanticDeprecatedSince20: Pydantic V1 style `@validator` validators are deprecated. You should migrate to Pydantic V2 style `@field_validator` validators, see the migration guide for more details. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.7/migration/
    @validator("model", pre=True)

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:341
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:341
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:341
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:341
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:341: PydanticDeprecatedSince20: Pydantic V1 style `@root_validator` validators are deprecated. You should migrate to Pydantic V2 style `@model_validator` validators, see the migration guide for more details. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.7/migration/
    @root_validator(skip_on_failure=True)

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:363
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:363
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:363
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:363
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\mlflow\gateway\config.py:363: PydanticDeprecatedSince20: Pydantic V1 style `@validator` validators are deprecated. You should migrate to Pydantic V2 style `@field_validator` validators, see the migration guide for more details. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.7/migration/
    @validator("route_type", pre=True)

..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pydantic\_internal\_config.py:284
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pydantic\_internal\_config.py:284
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pydantic\_internal\_config.py:284
..\..\..\..\..\AppData\Local\Programs\Python\Python311\Lib\site-packages\pydantic\_internal\_config.py:284
  C:\Users\mridu\AppData\Local\Programs\Python\Python311\Lib\site-packages\pydantic\_internal\_config.py:284: PydanticDeprecatedSince20: Support for class-based `config` is deprecated, use ConfigDict instead. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.7/migration/
    warnings.warn(DEPRECATION_MESSAGE, DeprecationWarning)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

---------- coverage: platform win32, python 3.11.9-final-0 -----------
Coverage HTML written to dir htmlcov
Coverage XML written to file coverage.xml

=============================================================================== slowest 20 durations ================================================================================
13.40s call     sktime/forecasting/personal/test1.py::test1.HFTransformersForecaster
0.03s setup    sktime/forecasting/personal/test1.py::test1.HFTransformersForecaster

(1 durations < 0.005s hidden.  Use -vv to show these durations.)
==================================================================== 1 passed, 88 warnings in 292.66s (0:04:52) ===================================================================== 
```

**Comment by Spinachboul** [comment]
@fkiraly  and @benHeid 
Could you please explain why this check keeps failing?
```{python}
Running Sphinx v7.3.7
making output directory... done
Adding stylesheet: fields.css
[autosummary] generating autosummary for: about.rst, about/_mission.rst, about/artwork.rst, about/citation.md, about/contributors.md, about/funding.rst, about/history.rst, about/mission.rst, about/showcase.rst, about/team.rst, ..., glossary.rst, includes/api_css.rst, index.rst, installation.rst, related_software.rst, roadmap.rst, tutorials.rst, user_guide.rst, user_guide/introduction.rst, users.rst
/home/docs/checkouts/readthedocs.org/user_builds/sktime/envs/6583/lib/python3.11/site-packages/sktime/utils/dependencies/_dependencies.py:293: UserWarning: `mlflow` is an extra dependency and is not included in the base sktime installation. Please run `pip install mlflow` or `pip install sktime[mlflow]` to install the package.
  return _check_soft_dependencies(

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/sktime/envs/6583/lib/python3.11/site-packages/sphinx/ext/autosummary/generate.py", line 503, in generate_autosummary_docs
    name, obj, parent, modname = import_by_name(entry.name)
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/docs/checkouts/readthedocs.org/user_builds/sktime/envs/6583/lib/python3.11/site-packages/sphinx/ext/autosummary/__init__.py", line 658, in import_by_name
    raise ImportExceptionGroup('no module named %s' % ' or '.join(tried), exceptions)
sphinx.ext.autosummary.ImportExceptionGroup: no module named sktime.forecasting.hf_transformers_forecaster.HFTransformersForecaster

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/sktime/envs/6583/lib/python3.11/site-packages/sphinx/events.py", line 97, in emit
    results.append(listener.handler(self.app, *args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/docs/checkouts/readthedocs.org/user_builds/sktime/envs/6583/lib/python3.11/site-packages/sphinx/ext/autosummary/__init__.py", line 818, in process_generate_options
    generate_autosummary_docs(genfiles, suffix=suffix, base_path=app.srcdir,
  File "/home/docs/checkouts/readthedocs.org/user_builds/sktime/envs/6583/lib/python3.11/site-packages/sphinx/ext/autosummary/generate.py", line 508, in generate_autosummary_docs
    name, obj, parent, modname = import_ivar_by_name(entry.name)
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/docs/checkouts/readthedocs.org/user_builds/sktime/envs/6583/lib/python3.11/site-packages/sphinx/ext/autosummary/__init__.py", line 715, in import_ivar_by_name
    real_name, obj, parent, modname = import_by_name(name, prefixes)
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/docs/checkouts/readthedocs.org/user_builds/sktime/envs/6583/lib/python3.11/site-packages/sphinx/ext/autosummary/__init__.py", line 658, in import_by_name
    raise ImportExceptionGroup('no module named %s' % ' or '.join(tried), exceptions)
sphinx.ext.autosummary.ImportExceptionGroup: no module named sktime.forecasting.hf_transformers_forecaster

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/sktime/envs/6583/lib/python3.11/site-packages/sphinx/cmd/build.py", line 332, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/docs/checkouts/readthedocs.org/user_builds/sktime/envs/6583/lib/python3.11/site-packages/sphinx/application.py", line 268, in __init__
    self._init_builder()
  File "/home/docs/checkouts/readthedocs.org/user_builds/sktime/envs/6583/lib/python3.11/site-packages/sphinx/application.py", line 339, in _init_builder
    self.events.emit('builder-inited')
  File "/home/docs/checkouts/readthedocs.org/user_builds/sktime/envs/6583/lib/python3.11/site-packages/sphinx/events.py", line 108, in emit
    raise ExtensionError(
sphinx.errors.ExtensionError: Handler <function process_generate_options at 0x7f2b163bdc60> for event 'builder-inited' threw an exception (exception: no module named sktime.forecasting.hf_transformers_forecaster)

Extension error (sphinx.ext.autosummary):
Handler <function process_generate_options at 0x7f2b163bdc60> for event 'builder-inited' threw an exception (exception: no module named sktime.forecasting.hf_transformers_forecaster)
```

**Comment by fkiraly** [comment]
looks like a broken import?

Here's a guide on ensuring code quality and how to automate this locally: https://www.sktime.net/en/stable/developer_guide/coding_standards.html

**Comment by Spinachboul** [comment]
@fkiraly and @benHeid 
could you please help me with the broken imports issue! I formatted code using `flake8` and `black` but this issue seems to persistent!

```{python}
Extension error (sphinx.ext.autosummary):
Handler <function process_generate_options at 0x7f81fc1bdc60> for event 'builder-inited' threw an exception (exception: no module named sktime.forecasting.hf_transformers_forecaster)
```

**Comment by fkiraly** [comment]
I think this is not doc related but python import related. It looks like somewhere you have an import of a module that does not exist.

Further, it appears you are making substantial changes to the module. Please ensure you are up to date with `main`, i.e., update your fork and merge main into your branch, then resolve any conflicts.

**Pull request closed without merging**